src/scripts/components/main/content-navigation/content-navigation.tsx

```diff
@@ -0,0 +1,168 @@
+import React, { FunctionComponent, useEffect, useState } from "react";
+
+import { getNavCoordinates } from "../../../libs/get-navigation-position";
+
+import { StoryList } from "../../../types/story-list";
+import { Link } from "react-router-dom";
+import {
+  useContentScrollHandlers,
+  useContentTouchHandlers,
+} from "./use-content-event-handlers";
+
+import cx from "classnames";
+
+import styles from "./content-navigation.module.css";
+import { FormattedMessage } from "react-intl";
+
+interface Props {
+  showContentList: boolean;
+  contents: StoryList;
+  setSelectedContentId: React.Dispatch<React.SetStateAction<string | null>>;
+  category: string | null;
+  className?: string;
+  isMobile: boolean;
+}
+
+const ContentNavigation: FunctionComponent<Props> = ({
+  category,
+  showContentList,
+  contents,
+  setSelectedContentId,
+  className,
+  isMobile,
+}) => {
+  const navigationRef = React.useRef<HTMLUListElement | null>(null);
+
+  const [currentIndex, setCurrentIndex] = useState(0);
+  const maxIndex = contents.length;
```

Comment:
```suggestion
  const maxIndex = contents.length -1;
```
Max index is `length - 1`, right? 🤔

Reply:
It's misleading.. .this refers to the number of entries, not the index. Let me change it